tests/run/nogil_conditional.pyx

```diff
@@ -0,0 +1,242 @@
+# mode: run
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+
+def test(int x):
+    """
+    >>> test(0)
+    110
+    """
+    with nogil(True):
+        x = f_nogil(x)
+        with gil(True):
+            x = f_gil(x)
```

Comment:
Note that this is not a function that _needs_ the GIL. It's one that _acquires_ the GIL itself, on entry. Just use a normal Python function instead.

Reply:
changed f_gil to normal Python function.